app/code/core/Mage/Directory/Helper/Data.php

```diff
@@ -107,14 +107,14 @@ public function __construct(array $args = array())
 
     /**
      * Retrieve region collection
-     * @param string|array|null $countryFilter
+     *
      * @return Mage_Directory_Model_Resource_Region_Collection
      */
-    public function getRegionCollection($countryFilter = null)
+    public function getRegionCollection()
     {
         if (!$this->_regionCollection) {
             $this->_regionCollection = Mage::getModel('directory/region')->getResourceCollection()
-                ->addCountryFilter($countryFilter)
+                ->addCountryFilter($this->getAddress()->getCountryId())
```

Comment:
hmm?

Reply:
Mentioned above ....

`$countryFilter` wasn't definded. I've tested this.

Reply:
see https://github.com/OpenMage/magento-lts/issues/713
changed again in https://github.com/OpenMage/magento-lts/pull/1644